pkg/cli/admin/nodeimage/create.go

```diff
@@ -0,0 +1,578 @@
+	flags.StringVarP(&o.OutputName, "output-name", "o", "node.iso", "The name of the output image.")
+	return cmd
+}
+
+func NewCreateOptions(streams genericiooptions.IOStreams) *CreateOptions {
+	return &CreateOptions{
+		IOStreams: streams,
+	}
+}
+
+type CreateOptions struct {
+	genericiooptions.IOStreams
+	SecurityOptions imagemanifest.SecurityOptions
+
+	Config         *rest.Config
+	Client         kubernetes.Interface
+	ConfigClient   configclient.Interface
+	FSys           fs.FS
+	RemoteExecutor exec.RemoteExecutor
+	CopyStrategy   func(*rsync.RsyncOptions) rsync.CopyStrategy
+
+	AssetsDir  string
+	OutputName string
+
+	factory                  kcmdutil.Factory
+	nodeJoinerImage          string
+	nodeJoinerNamespace      *corev1.Namespace
+	nodeJoinerServiceAccount *corev1.ServiceAccount
+	nodeJoinerRole           *rbacv1.ClusterRole
+	nodeJoinerPod            *corev1.Pod
+	nodeJoinerExitCode       int
+	rsyncRshCmd              string
+}
+
+func (o *CreateOptions) Complete(f kcmdutil.Factory, cmd *cobra.Command, args []string) error {
+	o.factory = f
+
+	var err error
+	if o.Config, err = f.ToRESTConfig(); err != nil {
+		return err
+	}
+	if o.Client, err = kubernetes.NewForConfig(o.Config); err != nil {
+		return err
+	}
+	if o.ConfigClient, err = configclient.NewForConfig(o.Config); err != nil {
+		return err
+	}
+
+	if o.AssetsDir == "" {
+		cwd, err := os.Getwd()
+		if err != nil {
+			return err
+		}
+		o.AssetsDir = cwd
+	}
+	o.FSys = os.DirFS(o.AssetsDir)
+	o.RemoteExecutor = &exec.DefaultRemoteExecutor{}
+	o.rsyncRshCmd = rsync.DefaultRsyncRemoteShellToUse(cmd)
+	o.CopyStrategy = func(o *rsync.RsyncOptions) rsync.CopyStrategy {
+		return rsync.NewDefaultCopyStrategy(o)
+	}
+	return nil
+}
+
+func (o *CreateOptions) Validate() error {
+	return nil
+}
+
+func (o *CreateOptions) Run(ctx context.Context) error {
+	defer o.cleanup(ctx)
+
+	err := o.runNodeJoinerPod(ctx)
+	if err != nil {
+		return err
+	}
+
+	err = o.waitForCompletion(ctx)
+	if err != nil {
+		return err
+	}
+	// Something went wrong during the node-joiner tool execution,
+	// let's show the logs and return an error
+	if o.nodeJoinerExitCode != 0 {
+		err = o.printLogs(ctx)
+		if err != nil {
+			return err
+		}
+		return fmt.Errorf("image generation error (exit code: %d)", o.nodeJoinerExitCode)
+	}
+
+	err = o.copyArtifactsFromNodeJoinerPod()
+	if err != nil {
+		return err
+	}
+	fmt.Println("Command successfully completed")
+	return nil
+}
+
+func (o *CreateOptions) printLogs(ctx context.Context) error {
+	logOptions := &corev1.PodLogOptions{
+		Container:  nodeJoinerContainer,
+		Timestamps: true,
+	}
+	readCloser, err := o.Client.CoreV1().Pods(o.nodeJoinerNamespace.GetName()).GetLogs(o.nodeJoinerPod.GetName(), logOptions).Stream(ctx)
+	if err != nil {
+		return err
+	}
+	defer readCloser.Close()
+
+	_, err = io.Copy(os.Stdout, readCloser)
```

Comment:
os.Stdout (and derivates) never be used directly. Instead you can pass `IOStream` and you can use in here `iostream.Out`